pytorch_lightning/utilities/device_parser.py

```diff
@@ -241,10 +240,16 @@ def _check_data_type(device_ids: Any) -> None:
         MisconfigurationException:
             If ``device_ids`` of GPU/TPUs aren't ``int``, ``str``, sequence of ``int`` or ``None``
     """
-    if device_ids is not None and (
-        not isinstance(device_ids, (int, str, MutableSequence, tuple)) or isinstance(device_ids, bool)
-    ):
-        raise MisconfigurationException("Device ID's (GPU/TPU) must be int, string or sequence of ints or None.")
+    msg = "Device IDs (GPU/TPU) must be an int, a string, a sequence of ints or None, but you passed"
+
+    if device_ids is None:
+        return
+    elif isinstance(device_ids, (MutableSequence, tuple)):
+        for id_ in device_ids:
+            if type(id_) is not int:
```

Comment:
```suggestion
            if not isinstance(id_, int):
```

Reply:
please be careful with bools. `isinstance(False, int)` is true. Which means we won't raise an error for bools with this suggestion, but we should.

Reply:
I agree with @awaelchli:

```python
In [4]: not isinstance(False, (int, str))
Out[4]: False

In [5]: type(False) not in (int, str)
Out[5]: True
```

After this suggestion, it might not fail for booleans. @Borda - please let me know if I'm wrong! :) Resolving this conversation for now.